3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```
@@ -17,29 +17,35 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
-function requestData(url) {
-  // TODO return a promise using `fetch()`
+const body = document.querySelector('body');
+
+async function requestData(url) {
+  const response = await fetch(url);
+  return response.json();
```

This function uses fetch to make a request to the provided URL and then immediately returns the result of calling json on the response. However, it doesn't check if the fetch request was successful before doing this. If the request fails due to a network error or if the server responds with an HTTP error status, this function will still try to parse the response as JSON, which will lead to a `SyntaxError` when the response body doesn't contain valid JSON. To fix this, you should check if the response is ok before parsing it as JSON. You should also throw an Error with the HTTP status and `statusText` as Error message.